Mage.Sets/src/mage/cards/c/CodsworthHandyHelper.java

```diff
@@ -0,0 +1,167 @@
+        Ability ability = new ActivateAsSorceryActivatedAbility(new CodsworthHandyHelperEffect(), new TapSourceCost());
+        ability.addTarget(new TargetPermanent(filter2));
+        ability.addTarget(new TargetControlledCreaturePermanent());
+        this.addAbility(ability);
+    }
+
+    private CodsworthHandyHelper(final CodsworthHandyHelper card) {
+        super(card);
+    }
+
+    @Override
+    public CodsworthHandyHelper copy() {
+        return new CodsworthHandyHelper(this);
+    }
+}
+
+class CodsworthHandyHelperManaBuilder extends ConditionalManaBuilder {
+
+    @Override
+    public ConditionalMana build(Object... options) {
+        return new CodsworthHandyHelperConditionalMana(this.mana);
+    }
+
+    @Override
+    public String getRule() {
+        return "Spend this mana only to cast Aura and/or Equipment spells";
+    }
+}
+
+class CodsworthHandyHelperConditionalMana extends ConditionalMana {
+
+    public CodsworthHandyHelperConditionalMana(Mana mana) {
+        super(mana);
+        addCondition(new CodsworthHandyHelperManaCondition());
+    }
+
+    private CodsworthHandyHelperConditionalMana(final CodsworthHandyHelperConditionalMana conditionalMana) {
+        super(conditionalMana);
+    }
+
+    @Override
+    public CodsworthHandyHelperConditionalMana copy() {
+        return new CodsworthHandyHelperConditionalMana(this);
+    }
+}
+
+class CodsworthHandyHelperManaCondition implements Condition {
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        if (source instanceof SpellAbility) {
+            Card card = game.getCard(source.getSourceId());
+            return card != null && (
+                    card.getSubtype(game).contains(SubType.AURA) || card.getSubtype(game).contains(SubType.EQUIPMENT)
+            );
+        }
+        return false;
+    }
+}
+class CodsworthHandyHelperEffect extends OneShotEffect {
+
+    CodsworthHandyHelperEffect() {
+        super(Outcome.Benefit);
+        staticText = "Attach target Aura or Equipment you control to target creature you control";
+    }
+
+    private CodsworthHandyHelperEffect(final CodsworthHandyHelperEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public CodsworthHandyHelperEffect copy() {
+        return new CodsworthHandyHelperEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        Permanent attachment = game.getPermanent(source.getTargets().get(0).getFirstTarget());
+        Permanent creature = game.getPermanent(source.getTargets().get(1).getFirstTarget());
+        if (controller == null || attachment == null || creature == null) {
+            return false;
+        }
+
+        if (creature.cantBeAttachedBy(attachment, source, game, true)) {
+            game.informPlayers(attachment.getLogName() + " was not attached to " + creature.getLogName()
```

Comment:
You can add `CardUtil.getSourceLogName(game, source)` at the end of most log.
Adds info for the source (Codsworth), and when you hover over it, it show the card image, which is nice.

Reply:
Added in 30d14e0